src/components/avatar/image.js

```diff
@@ -1,203 +1,48 @@
-    this.loadImg();
-  };
-
-  loadImg = () => {
-    this.i = new Image();
-    this.i.src = this.sourceList[this.state.currentIndex];
-
-    if (this.props.decode && this.i.decode) {
-      this.i
-        // $FlowIssue
-        .decode()
-        .then(this.onLoad)
-        .catch(this.onError);
-    } else {
-      this.i.onload = this.onLoad;
-    }
-
-    this.i.onerror = this.onError;
-  };
-
-  unloadImg = () => {
-    try {
-      delete this.i.onerror;
-      delete this.i.onload;
-      delete this.i.src;
-      delete this.i;
-    } catch (err) {
-      console.error(err);
-      delete this.i;
-    }
-  };
-
-  componentDidMount() {
-    // kick off process
-    /* istanbul ignore else */
-    if (this.state.isLoading) this.loadImg();
-  }
-
-  componentDidUpdate() {
-    const curr = this.props;
-
-    let src = this.srcToArray(curr.src);
-
-    let srcAdded = src.filter(s => this.sourceList.indexOf(s) === -1);
-    let srcRemoved = this.sourceList.filter(s => src.indexOf(s) === -1);
-
-    // if src prop changed, restart the loading process
-    if (srcAdded.length || srcRemoved.length) {
-      this.sourceList = src;
-
-      // if we dont have any sources, jump directly to unloader
-      if (!src.length)
-        return this.setState({ isLoading: false, isLoaded: false });
-      return this.setState({
-        currentIndex: 0,
-        isLoading: true,
-        isLoaded: false,
-      });
-    }
-  }
-
-  componentWillUnmount() {
-    // ensure that we dont leave any lingering listeners
-    /* istanbul ignore else */
-    if (this.i) this.unloadImg();
-  }
-
+export default class Image extends React.Component<Props> {
   render() {
-    // if we have loaded, show img
-    if (this.state.isLoaded) {
-      // clear non img props
-      let { src, loader, unloader, decode, ...rest } = this.props; //eslint-disable-line
-      return (
-        <Img src={this.sourceList[this.state.currentIndex]} {...rest} alt="" />
-      );
-    }
-
-    // if we are still trying to load, show img and a loader if requested
-    if (!this.state.isLoaded && this.state.isLoading)
-      return <ImgPlaceholder {...this.props} />;
-
-    // if we have given up on loading, show a place holder if requested, or nothing
-    /* istanbul ignore else */
-    if (!this.state.isLoaded && !this.state.isLoading)
-      return this.props.unloader ? (
-        this.props.unloader
-      ) : (
-        <ImgPlaceholder {...this.props} />
-      );
+    const { type, size, mobilesize } = this.props;
+    const fallbackSrc =
+      type === 'user'
+        ? '/img/default_avatar.svg'
+        : '/img/default_community.svg';
+
+    return (
+      <VisibilitySensor>
+        <Img
+          {...this.props}
+          decode={false}
+          loader={
+            <Fallback
+              size={size}
+              mobilesize={mobilesize}
+              type={type}
+              src={fallbackSrc}
+              alt=""
+            />
```

Comment:
Why don't we just show nothing while loading the avatars? I think that'd make it feel more like the native image loading of the browser, it feels kinda weird to show the default while loading?

Reply:
Cleaned this up to add a subtle placeholder, agree on flashing the default being weird